docs/contribute/doc.md

```diff
@@ -0,0 +1,116 @@
+# Documentation style
+
+We are in the process of implementing new documentation requirements for mathlib. All future pull
+requests must meet the following standards.
+
+## Header comment
+
+Each mathlib file should start with a header comment with copyright information (see example below)
+followed by general documentation written using Markdown. Headers use atx-style headers (with hash
+signs, no underlying dash).
+
+The mandatory title of the file is a first level header. It is followed by a summary of the content
+of the file.
+
+The other sections, with second level headers are (in this order):
+* *Main definitions* (optional, can be covered in the summary)
+* *Main statements* (optional, can be covered in the summary)
+* *Notations* (ommited only if no notation is introduced in this file)
+* *Implementation notes* (description of important design decisions or interface features)
+* *References* (references to textbooks or papers, or Wikipedia pages)
+
+After references, we write "Tags:" followed by a list of keywords that could be useful when
+doing text search in mathlib to find where something is covered.
+
+The following code block is an example of a file header.
+
+```
+/-
+Copyright (c) 2018 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert Y. Lewis
+
+# p-adic norm
+
+This file defines the p-adic valuation and the p-adic norm on ℚ.
+
+The p-adic valuation on ℚ is the difference of the multiplicities of `p` in the numerator and
+denominator of `q`. This function obeys the standard properties of a valuation, with the appropriate
+assumptions on p.
+
+The valuation induces a norm on ℚ. This norm is a nonarchimedean absolute value.
+It takes values in {0} ∪ {1/p^k | k ∈ ℤ}.
+
+## Notations
+
+This file uses the local notation `/.` for `rat.mk`.
+
+## Implementation notes
+
+Much, but not all, of this file assumes that `p` is prime. This assumption is inferred automatically
+by taking (prime p) as a type class argument.
+
+## References
+
+* F. Q. Gouêva, *p-adic numbers*
```

Comment:
Thinking about search mechanisms, I would prefer a more structured, bibtex-like list of references (maybe like [this](https://github.com/kappelmann/lean-continued-fractions/blob/master/src/definitions/continued_fractions.lean#L32))

Reply:
While this citation could be expanded a bit, I'm not sure that going to bibtex is the right approach. If we were doing real semantic tagging, maybe. But this is just a string at the top of the file that will be read by people, not automatically parsed, and nobody is going to search mathlib for publisher names or ISBN numbers.

Reply:
Long term I would like this to be a link to a bibliography in some centralized location. The citation might be a latex style `\cite{Gouêva}`, or maybe something more markdown-friendly, but I think we definitely want these references to be computer readable so that we can later link back to this location if we are looking for theorems that use the book and so on.

Reply:
I like the central database more than inline bibtex. (This also avoids multiple files citing variations of the same reference.) Something like `[Gouêva](F. Q. Gouêva, *p-adic numbers*)` would be nice. It's readable inline, without too much extra stuff around it, and refers to some database entry tagged `Gouêva`. But this parses as a markdown link and I'm not sure how to set things up to make it a valid one. It's hard to make a concrete proposal here without knowing how these will be parsed and for what purpose.

Reply:
Since there's no use case yet, I propose we go with `[Gouêva](F. Q. Gouêva, *p-adic numbers*)`, referencing a central bibtex file. Support for bibliographies in markdown seems to be pretty inconsistent, but this seems future-proof enough to automatically change to another format if/when necessary.

Reply:
Actually, `[F. Q. Gouêva, *p-adic numbers*][gouvea1997]` seems to be closer to a markdown-style reference.